src/Kiota.Builder/Writers/CSharp/CSharpConventionService.cs

```diff
@@ -2,49 +2,74 @@
     public class CSharpConventionService : CommonLanguageConventionService {
         public override string StreamTypeName => "stream";
         public override string VoidTypeName => "void";
         public override string DocCommentPrefix => "/// ";
-        public override string PathSegmentPropertyName => "PathSegment";
-        public override string CurrentPathPropertyName => "CurrentPath";
-        public override string HttpCorePropertyName => "HttpCore";
-        public HashSet<string> NullableTypes { get; } = new() { "int", "bool", "float", "double", "decimal", "Guid", "DateTimeOffset" };
-        public static string NullableMarker => "?";
+        private const string PathSegmentPropertyName = "PathSegment";
+        private const string CurrentPathPropertyName = "CurrentPath";
+        private const string HttpCorePropertyName = "HttpCore";
+        public HashSet<string> NullableTypes { get; } = new(StringComparer.OrdinalIgnoreCase) { "int", "bool", "float", "double", "decimal", "long", "Guid", "DateTimeOffset" };
+        public static readonly char NullableMarker = '?';
+        public static string NullableMarkerAsString => "?";
         public override string ParseNodeInterfaceName => "IParseNode";
-        public override string RawUrlPropertyName => "IsRawUrl";
         public override void WriteShortDescription(string description, LanguageWriter writer) {
             if(!string.IsNullOrEmpty(description))
                 writer.WriteLine($"{DocCommentPrefix}<summary>{description}</summary>");
         }
         public override string GetAccessModifier(AccessModifier access)
         {
-            return (access) switch {
-                (AccessModifier.Public) => "public",
-                (AccessModifier.Protected) => "protected",
+            return access switch {
+                AccessModifier.Public => "public",
+                AccessModifier.Protected => "protected",
                 _ => "private",
             };
         }
-        internal void AddRequestBuilderBody(bool addCurrentPath, string returnType, LanguageWriter writer, string suffix = default, string prefix = default) {
+        #pragma warning disable CA1822 // Method should be static
+        internal void AddRequestBuilderBody(bool addCurrentPath, string returnType, LanguageWriter writer, string suffix = default, string prefix = default, IEnumerable<CodeParameter> pathParameters = default) {
             var currentPath = addCurrentPath ? $"{CurrentPathPropertyName} + " : string.Empty;
-            writer.WriteLine($"{prefix}new {returnType}({currentPath}{PathSegmentPropertyName} {suffix}, {HttpCorePropertyName}, false);");
+            var pathParametersSuffix = !(pathParameters?.Any() ?? false) ? string.Empty : $"{string.Join(", ", pathParameters.Select(x => $"{x.Name}"))}, ";
+            writer.WriteLine($"{prefix}new {returnType}({currentPath}{PathSegmentPropertyName} {suffix}, {HttpCorePropertyName}, {pathParametersSuffix}false);");
```

Comment:
Just a thought, but there is the potential to omit the `{returnType}` if we know the caller already defined the type on the left hand side of the declaration.

Reply:
This is an interesting suggestion. This feature was introduced [by C# 9](https://docs.microsoft.com/en-us/dotnet/csharp/whats-new/csharp-9#fit-and-finish-features) and although I believe it'd only compile time sugar (no runtime requirements on the VM), [it's not clearly stateed](https://docs.microsoft.com/en-us/dotnet/csharp/whats-new/csharp-version-history#c-version-90) and pushes a requirement of targeting dotnet 5 onto the user.
We haven't made a clear decision on what compiler version and target runtime/aspi spec we're going to push on Kiota users, but I'm pretty sure what Kiota is generating so far doesn't require anything more than C# 6, which would give us a pretty broad compatibility.

Reply:
I suspect that we would push down the requirement as well. I'm happy to keep as is to be on the safe side.